stdlib/stdlib.ispc

```diff
@@ -520,128 +506,114 @@ template <typename T> unmasked varying T __shift_via_shuffle(varying T v, unifor
     return select(inBounds, shuffled, (varying T)0);
 }
 
-#if defined(ISPC_TARGET_AVX512SKX) || defined(ISPC_TARGET_AVX512ICL) || defined(ISPC_TARGET_AVX512SPR) ||              \
-    defined(ISPC_TARGET_AVX10_2)
-#define USE_SHIFT_VIA_SHUFFLE_8
-#define USE_SHIFT_VIA_SHUFFLE_16
-
-#if TARGET_WIDTH < 64
-#define USE_SHIFT_VIA_SHUFFLE_32
-#endif
-
-#if TARGET_WIDTH < 32
-#define USE_SHIFT_VIA_SHUFFLE_64
-#endif
-#endif // avx512 or avx10
-
 __declspec(safe) static inline float shift(float v, uniform int i) {
-#ifdef USE_SHIFT_VIA_SHUFFLE_32
-    return __shift_via_shuffle(v, i);
-#else
-    varying float result;
-    unmasked { result = __shift_float(v, i); }
-    return result;
-#endif
+    if (__have_shift_via_shuffle_32) {
```

Comment:
It doesn't feel right to me to have `__have_shift_via_shuffle_32` treated as something similar to a capability. It seems to me it would be better to move these implementations, along with the conditional logic, into the generic builtins implementation.

Reply:
I agree that `__have_shift_via_shuffle_32` is not a capability.

Do you mean do something like this? https://github.com/aneshlya/ispc/commit/ab765afec4d09031421717542e4404dbd53c17bd
I'm not sure how to make it work because when we compile `generic.ispc` ISPC_TARGET_ macro is not set.